File: src/java/magma/compile/rule/StripRule.java
package magma.compile.rule;

import magma.compile.CompileError;
import magma.compile.Node;
import magma.result.Result;

public record StripRule(Rule childRule) implements Rule {
    @Override
    public Result<Node, CompileError> parse(String input) {
        return childRule().parse(input.strip());
    }

    @Override
    public Result<String, CompileError> generate(Node node) {
        return childRule.generate(node);
    }
}

File: src/ts/magma/app/compile/rule/StripRule.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { Rule } from "../../../../magma/app/compile/rule/Rule";
import { Option } from "../../../../magma/api/option/Option";
import { Strings } from "../../../../magma/api/text/Strings";
export class StripRule implements Rule<Node> {
	childRule: Rule<Node>;
	constructor (childRule: Rule<Node>) {
		this.childRule = childRule;
	}
	lex(input: string): Option<Node> {
		return this.childRule.lex(Strings.strip(input))/*unknown*/;
	}
}
